docs/development/CI/infrastructure.md

```diff
@@ -4,20 +4,111 @@
 
 The following workflows are currently active:
 
-![workflows](workflows.drawio.svg)
-
-The `Eve / Test` and `Gt4py / Test` workflows run the automated tests for the two packages. These workflows also save the code coverage output as workflow artifacts.
-
-The `Eve / Coverage` and `Gt4py / Coverage` workflows are triggered by a successful run of the tests. They download the coverage artifacts and publish them to codecov.io.
-
-The `Documentation` workflow executes the Jupyter nodebook of the quick start guide to make sure it's in sync with the code.
+<!--
+@startuml workflows
+
+left to right direction
+skinparam linetype ortho
+
+'!include <C4/C4_Component>
+
+card always [
+always
+]
+file src_eve #Plum;line:MediumOrchid;text:black [
+eve sources changed
+]
+file src_car #Gold;line:DarkGoldenRod;text:black [
+cartesian sources changed
+]
+file src_nxt #APPLICATION;line:CornFlowerBlue;text:black [
+next sources changed
+]
+file src_sto #Aquamarine;line:DarkCyan;text:black [
+storage sources changed
+]
+file src_cab #Gold;line:DarkGoldenRod;text:black [
+cartesian backend sources changed
+]
+file cfg_wfl #Coral;line:Tomato;text:black[
+workflows changed
+]
+file other #PaleGreen;line:ForestGreen;text:black [
+other files changed, excluding examples (the examples folder) and docs (.md and .rst files)
+]
+file pkg_cfg #PaleGreen;line:ForestGreen;text:black [
+package config files changed
+]
+
+agent eve #Plum;line:MediumOrchid;text:black [
+Test / Eve
+]
+agent car #Gold;line:DarkGoldenRod;text:black [
+Test / Cartesian (CPU)
+]
+agent nxt #APPLICATION;line:CornFlowerBlue;text:black [
+Test / Next (CPU)
+]
+agent sto #Aquamarine;line:DarkCyan;text:black [
+Test / Storage (CPU)
+]
+agent qua [
+Code Quality
+]
+
+always  -[hidden]r-> src_eve
+src_eve -[hidden]r-> src_car
+src_car -[hidden]r-> src_nxt
+src_nxt -[hidden]r-> src_sto
+src_sto -[hidden]r-> src_cab
+src_cab -[hidden]r-> cfg_wfl
+cfg_wfl -[hidden]r-> other
+
+always  -d-> qua
+
+src_eve -[#MediumOrchid,thickness=2]d-> car
+src_car -[#DarkGoldenRod,thickness=2]d-> car
+src_sto -[#DarkCyan,thickness=2]d-> car
+cfg_wfl -[#Tomato,thickness=2]d-> car
+other   -[#ForestGreen,thickness=2]d-> car
+
+src_eve -[#MediumOrchid,thickness=2]d-> eve
+cfg_wfl -[#Tomato,thickness=2]d-> eve
+pkg_cfg -[#ForestGreen,thickness=2]d-> eve
+
+src_eve -[#MediumOrchid,thickness=2]d-> nxt
+src_nxt -[#CornFlowerBlue,thickness=2]d-> nxt
+src_sto -[#DarkCyan,thickness=2]d-> nxt
+cfg_wfl -[#Tomato,thickness=2]d-> nxt
+other   -[#ForestGreen,thickness=2]d-> nxt
+
+src_sto -[#DarkCyan,thickness=2]d-> sto
+src_cab -[#DarkGoldenRod,thickness=2]d-> sto
+cfg_wfl -[#Tomato,thickness=2]d-> sto
+pkg_cfg -[#ForestGreen,thickness=2]d-> sto
+
+@enduml
+-->
+![](workflows.svg)
```

Comment:
Just a comment: have you considered to use mermaid (https://mermaid.live)? We could embed the source diagram in the markdown file as you do here, but it would allow us to easily edit it online (and it probably also look better?

Reply:
The test boxes in the diagram still use the old `Test / Something` naming (with the slash in the middle)

Reply:
Check the results here: https://github.com/DropD/gt4py/blob/ci-docs/docs/development/CI/infrastructure.md. It seems to have trouble rendering it, although mermaid does so fine: [![](https://mermaid.ink/img/pako:eNqdlu1O2zAUhm_FilQVJEArDNiCNAnKQEgDusG0H02FXNtJrTp25zi0VcW979itS5Jmakqr5sN-n9fH59iVFwFRlAVh0GotEJfchGiB2rFQUzLC2rTdK8n1K4PHtsQm11i00Zv9tlqRXCvRj1-RRPDBYornWX95G6DDw2_ob477XRgG_cyx4GY-iORSm2nywl5ZH34oU7kmLEPgJhNGl6TtjIJnlhn0_ZVFwaDEOQnB2ku6EAfLOJZor9v7vV8rlzPj5Q9sZgrKd621JGuv2sC2jFr0Gha8hpiMmaT1nplR3vPJKI0TVutoZdmq_38-JbUPGG22QjaK1jY50mal1rcgJnHyMo1Ff6r02C6BzeygktAXc7PRmVYb3QyWzcqMmE21u6OYi_eoDhCbEZFTLhN4wunE9u2B7v0tVoIyvY8w5JwqAt1HKXVvRxrS7Oxsfn3UhTGrU56MkxeIsT-BEtrUEyVjnpQjGpTnuUKqU8rMXLDVPrG8CP-MuGEHmdFqzMKhgBEuilK_fp22J_LUS-8Z5Xn6qMmI0w3CFsERt5AET1xpNZU10mFTqV05TnoJuzrFmst14NdYj7tzvMlAFldMr_fjrnv5fPf44KGu0vIGVhDTVyJnJdSvCIeCDgsPPasUG1USr9cHZAgLdqsZk15-ozRsKtdUYnx9mlBFDmbeqG671axpvXYswM7JX14Fl-MnZ9BBNZGvHA6nnJpReDyZXVSx449hJ02xKvgZFVO2faDTrfoqcYYqed4-yPnuyJcmSBX6imor2gTtfEKlXbU9wk5nZ-J4Z-JkO7HBrJdAYes2GOr0Y9jZx7DzRlhwEKRMp5hTOJstrEkUwD9cCmegEB4pi3EuTBRE8g2kODfqaS5JEBoNVQ_yCcWGXXOcaJwGYYxFBq2woeDYcL8877lj39s_ILp1lw?type=png)](https://mermaid.live/edit#pako:eNqdlu1O2zAUhm_FilQVJEArDNiCNAnKQEgDusG0H02FXNtJrTp25zi0VcW979itS5Jmakqr5sN-n9fH59iVFwFRlAVh0GotEJfchGiB2rFQUzLC2rTdK8n1K4PHtsQm11i00Zv9tlqRXCvRj1-RRPDBYornWX95G6DDw2_ob477XRgG_cyx4GY-iORSm2nywl5ZH34oU7kmLEPgJhNGl6TtjIJnlhn0_ZVFwaDEOQnB2ku6EAfLOJZor9v7vV8rlzPj5Q9sZgrKd621JGuv2sC2jFr0Gha8hpiMmaT1nplR3vPJKI0TVutoZdmq_38-JbUPGG22QjaK1jY50mal1rcgJnHyMo1Ff6r02C6BzeygktAXc7PRmVYb3QyWzcqMmE21u6OYi_eoDhCbEZFTLhN4wunE9u2B7v0tVoIyvY8w5JwqAt1HKXVvRxrS7Oxsfn3UhTGrU56MkxeIsT-BEtrUEyVjnpQjGpTnuUKqU8rMXLDVPrG8CP-MuGEHmdFqzMKhgBEuilK_fp22J_LUS-8Z5Xn6qMmI0w3CFsERt5AET1xpNZU10mFTqV05TnoJuzrFmst14NdYj7tzvMlAFldMr_fjrnv5fPf44KGu0vIGVhDTVyJnJdSvCIeCDgsPPasUG1USr9cHZAgLdqsZk15-ozRsKtdUYnx9mlBFDmbeqG671axpvXYswM7JX14Fl-MnZ9BBNZGvHA6nnJpReDyZXVSx449hJ02xKvgZFVO2faDTrfoqcYYqed4-yPnuyJcmSBX6imor2gTtfEKlXbU9wk5nZ-J4Z-JkO7HBrJdAYes2GOr0Y9jZx7DzRlhwEKRMp5hTOJstrEkUwD9cCmegEB4pi3EuTBRE8g2kODfqaS5JEBoNVQ_yCcWGXXOcaJwGYYxFBq2woeDYcL8877lj39s_ILp1lw)

Reply:
As for better looking, the link styles provided by mermaid don't seem to work as well for this. Otherwise it seems just equivalent.